explorer/e2e/testFunctions.ts

```diff
@@ -301,27 +301,39 @@ export async function testFilterPresence(
  * @param filterOptionName - the name of the filter option
  * @returns a Playwright locator to the filter button
  */
-export const getNamedFilterButtonLocator = (
+export const getNamedFilterOptionLocator = (
   page: Page,
   filterOptionName: string
 ): Locator => {
-  return page
-    .getByRole("button")
-    .filter({ has: page.getByRole("checkbox"), hasText: filterOptionName });
+  return page.getByRole("button").filter({
+    has: page.getByRole("checkbox"),
+    hasText: RegExp(`^${escapeRegExp(filterOptionName)}\\s*\\d+\\s*`),
+  });
 };
 
 /**
  * Get a locator for the first filter option. Requires a filter menu to be open
  * @param page - a Playwright page object
  * @returns a Playwright locator to the filter button
  */
-export const getFirstFilterButtonLocator = (page: Page): Locator => {
+export const getFirstFilterOptionLocator = (page: Page): Locator => {
   return page
     .getByRole("button")
     .filter({ has: page.getByRole("checkbox") })
     .first();
 };
 
+export const getFilterOptionName = async (
```

Comment:
Could you add TSDocs for this function, and add a bit of detail around the [split/find](https://github.com/DataBiosphere/data-browser/pull/4135/files#diff-faefc5cfc7d7c7283d514ad866678143558aefea3eccb636b8bb7a1ae4abbae7R332-R333) functionality?